### PR TITLE
Command history

### PR DESCRIPTION
**Pull request by kevinlmadison**
Fixed the command history scrolling

### CODE REVIEW

File: src/screen.rs [hunk 1]
```diff
@@ -67,6 +71,8 @@ impl Interfaceable for PrettyPrompt {
             }
             Err(_) => String::from("quit"),
         };
+        self.editor.save_history(".history.txt").unwrap();
```
Comment (dmadisetti) [inline src/screen.rs]: This should only run if history is true

File: src/screen.rs [hunk 2]
```diff
@@ -7,6 +7,8 @@
 extern crate rustyline;
 #[cfg(feature = "pretty")]
 use self::rustyline::Editor;
+use std::fs::File;
```
Comment (dmadisetti) [inline src/screen.rs]: I be this gives unused deps when compiling normally right

File: src/screen.rs [hunk 3]
```diff
@@ -42,6 +44,9 @@ pub struct PrettyPrompt {
 impl Interfaceable for PrettyPrompt {
     fn new() -> PrettyPrompt {
         let mut rl = Editor::<()>::new();
+        if !Path::new(".history.txt").exists() {
```
Comment (dmadisetti) [inline src/screen.rs]: Low-key, is there a way to do history without a file?

File: src/screen.rs (hunk 1 above)
Comment (dmadisetti) [reply src/screen.rs]: Also you should unwrap safely

**Review by dmadisetti — COMMENTED** [review]
Aside from that, it works out of the box? Awesome

File: src/screen.rs (hunk 2 above)
Comment (dmadisetti) [reply src/screen.rs]: *bet

File: src/screen.rs (hunk 1 above)
Comment (kevinlmadison) [reply src/screen.rs]: Word, I was thinking I might need to put a check there.

File: src/screen.rs (hunk 2 above)
Comment (kevinlmadison) [reply src/screen.rs]: It does, should I add an allod_unused_imports macro?

File: src/screen.rs (hunk 3 above)
Comment (kevinlmadison) [reply src/screen.rs]: Twould be nice, I'll see if I can figure that out.

File: src/screen.rs [hunk 4]
```diff
@@ -97,3 +107,16 @@ pub type Screen = BasicPrompt;
...
 #[cfg(feature = "pretty")]
 pub type Screen = PrettyPrompt;
+
+#[cfg(feature = "pretty")]
+fn confirm_history() -> bool {
+    if !Path::new(".history.txt").exists() {
+        match File::create(Path::new(".history.txt")) {
+            Ok(_) => return true,
+            Err(err) => {
+                println!("Error: {:?}", err);
```
Comment (dmadisetti) [inline src/screen.rs:117]: Don't use println, use self.print

Comment (dmadisetti) [reply src/screen.rs:117]: Make confirm history part of the pretty struct

File: src/screen.rs [hunk 5]
```diff
@@ -67,6 +70,13 @@ impl Interfaceable for PrettyPrompt {
             }
             Err(_) => String::from("quit"),
         };
+        if confirm_history() {
+            match self.editor.save_history(".history.txt") {
+                Ok(_) => (),
+                Err(_) => return String::from("Error writing .history.txt."),
```
Comment (dmadisetti) [inline src/screen.rs:76]: Return nothing. Otherwise we'll try to parse this

File: src/screen.rs [hunk 6]
```diff
@@ -7,6 +7,10 @@
 extern crate rustyline;
```
Comment (dmadisetti) [inline src/screen.rs:7]: Add .history.txt to gitignore

**Comment by dmadisetti** [comment]
Did my comments go through?

**Review by dmadisetti — COMMENTED** [review]
Ahh woops. Forgot to hit submit

**Pull request merged**